src/Platforms/SYCL/SYCLDeviceManager.h

```diff
@@ -0,0 +1,50 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2022 QMCPACK developers.
+//
+// File developed by: Thomas Applencourt, apl@anl.gov, Argonne National Laboratory
+//                    Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//
+// File created by: Thomas Applencourt, apl@anl.gov, Argonne National Laboratory
+//
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_SYCLDEVICEMANAGER_H
+#define QMCPLUSPLUS_SYCLDEVICEMANAGER_H
+
+#include <CL/sycl.hpp>
+#include <vector>
+#include "config.h"
+
+namespace qmcplusplus
+{
+
+struct syclDeviceInfo
+{
+  sycl::context context;
+  sycl::device device;
+};
+
+#if defined(_OPENMP)
+std::vector<struct syclDeviceInfo> xomp_get_sycl_devices();
```

Comment:
does this need to be in the global qmcplusplus namespace?

Reply:
Move it to cpp and marked it static. It is only needed in the file scope.